[ZIP TBD]: draft-coinholder-directed-grants.md (#1112)
zcash/zips: zips/draft-coinholder-directed-grants.md

* reducing administrative complexity for Key-Holder Organizations;  
* enabling direct, transparent coinholder participation; and  
* ensuring that funds are only disbursed for measurable contributions that advance Zcash.  

---

## Requirements

1. **Retroactive Funding Model** — Grants MUST be awarded solely for work that is demonstrably complete and publicly verifiable.  
2. **Coinholder Voting Process** — All grants MUST be approved by a coinholder vote meeting the participation threshold defined in ZIP 1016 (≥ 420,000 ZEC voted).  
3. **Transparency and Reporting** — Proposal information, vote outcomes, and disbursements MUST be published publicly by the Administrator or its delegate.  
4. **Administrative Safeguards** — The Administrator or its delegate SHALL manage proposal submission, eligibility screening, compliance checks, and quarterly reporting, including KYC for grants > 50,000 USD equivalent.  
5. **Keyholder Oversight** — Key-Holder Organizations MUST execute approved disbursements unless vetoed under the conditions defined in ZIP 1016 (Veto Process).  
6. **Legal Compliance** — All parties MUST comply with applicable legal, tax, and reporting obligations.  

> 7. **Security and Custody** — Multisig keys MUST be secured per the standards in ZIP 1016.  

Reviewer: This content should be in `## Specification`.

The Requirements section is explicitly to define what requirements _this ZIP_ must satisfy; i.e. constraints on this specification itself, or on the author of the specification. The Requirements section should not impose any conformance requirements. The requirements section guides the authoring of the specification; it is not specification itself.

---

## Non-Requirements

* Changes to governance or funding mechanisms of other programs are **outside scope**.  
* This ZIP does not modify consensus parameters or introduce new Funding Streams beyond those defined in ZIP 1016.  

---

## Specification

### 1. Program Structure

> The Coinholder Grants Program SHALL operate as a **retroactive funding mechanism** for the 12% Funding Stream defined in ZIP 1016.  

Reviewer: See https://github.com/zcash/zips/pull/1112/files#r2582941120

Funds are held in the Coinholder-Controlled Fund multisig wallet and distributed per approved proposals.  

> The Administrator or its delegate is responsible for managing proposal intake, eligibility review, publication, and transparency reporting.  

Reviewer: TBD will be the Administrator of the Coinholder-Controlled Fund.

Key-Holder Organizations (ECC, ZF, Shielded Labs) execute multisig disbursements upon coinholder approval, subject to veto provisions.  

---

### 2. KYC and Compliance

> Grants > 50,000 USD require KYC verification of the recipient by the Administrator or its delegate prior to payment.  

Reviewer: This needs to be stated as a conformance requirement.

The Administrator or its delegate MAY charge a reasonable administrative fee to cover these duties, subject to coinholder approval.  

---

### 3. Participation and Voting Thresholds

* Participation threshold: ≥ 420,000 ZEC voted (≈ 2% of total supply).  
* Approval criterion: simple majority of “Yes” votes.  
* Each proposal vote is independent; coins used in one vote MAY be reused in another.  

---

### 4. Voting Process

The Coinholder Grants Program supports two voting methods: **Shielded Voting** and **Transparent Voting**.  

Both MUST produce verifiable and auditable results administered by the Administrator or its delegate.  

#### 4.1 Shielded Voting Process

Shielded voting is conducted through the dedicated Coin Voting 2.0 application that uses Orchard addresses and the Halo2 proof system to verify balances and prevent double voting.  

1. **Registration Period** – The Administrator or its delegate MUST define a registration window not shorter than 10 days, specified by block heights. Only Orchard notes created or refreshed within this window are eligible for voting.  

> 2. **Pre-Vote Safeguards** – Participants SHOULD move funds out of the wallet used for voting *before* entering their seed phrase into the Coin Voting application to reduce risk if the third-party app is compromised.  

Reviewer: Participants aren't bound by this spec, but maybe something about "the Administrators MUST provide clear guidance to participants ..."

3. **Voting Phase** – After registration closes, eligible coinholders vote using their Orchard funds via the Coin Voting application. Each ZEC held corresponds to 1,000 vote units.  
4. **Vote Submission** – Votes are submitted to a dedicated voting blockchain operated by independent Voting Authorities running a Byzantine Fault Tolerant (BFT) consensus network (e.g., CometBFT).  
5. **Tally and Audit** – At the end of the voting period, the Administrator or its delegate collects finalized results from the Voting Authorities. Auditors MAY independently verify the tally using the public viewing key and revealed election seed.  
6. **Security Assumptions** – The system MUST include ≥ 4 Voting Authorities to ensure liveness and integrity; so long as ≥ ⅔ are honest, the vote cannot be manipulated.  

#### 4.2 Transparent Voting Process

Transparent voting MAY be used alongside shielded voting to expand participation and maintain verifiability.  

> 1. Voters create a signed message containing their vote choice and transparent address.  

Reviewer: Need to specify signature scheme.

Reviewer: The coins need to be unspent as of the end of the registration period.

2. The signed message and a payment of ≥ 1 ZEC MUST be sent to a designated Orchard address controlled by the Administrator or its delegate.  

> 3. The Administrator or its delegate publishes the corresponding viewing key so anyone can verify vote contents and tally independently.  

Reviewer: This should go before step 2, as the IVK (is this intended to be and IFK or an FVK?) is what the address should be derived from; specify that derivation.

Author: Addressed, please see [97efc59](https://github.com/zcash/zips/pull/1112/commits/97efc598c945c5a4a64bfbf30f3a7d36f0d90d87)

4. Votes with payments < 1 ZEC MUST be ignored to mitigate Sybil attacks.  

> 5. Funds MUST remain immobile until the voting period concludes to prevent vote tampering.  

Author: Addressed, please see [97efc59](https://github.com/zcash/zips/pull/1112/commits/97efc598c945c5a4a64bfbf30f3a7d36f0d90d87)

> Both methods MUST produce publicly auditable vote records and a final summary published by the Administrator or its delegate within 7 days of vote conclusion.  

Reviewer: The Administrator must produce a publicly auditable vote record; whether the method produces an auditable vote record is something that the reviewers/ZIP editors will check.

The need to produce a publicly auditable vote record is part of the requirements.

---

### 5. Proposal Funding and Disbursement

Following the publication of voting results, approved proposals SHALL be funded in order of their relative voting support.  

1. **Prioritization by Vote Share** — Approved grants are ranked from highest to lowest based on the percentage of “Yes” votes received. Funding is distributed sequentially from the top of this ranking until available treasury funds for the quarter are fully allocated.  

> 2. **Partial Funding** — If the Coinholder Grants Program treasury cannot fully fund all approved grants within the same quarter, the remaining balance MAY be used to partially fund the next-ranked proposal. Partial funding MUST be proportional to the remaining available balance.  

Reviewer: Does this mean "the remaining balance after the higher-ranked proposals have been fully funded"?

Author: Addressed, please see [97efc59](https://github.com/zcash/zips/pull/1112/commits/97efc598c945c5a4a64bfbf30f3a7d36f0d90d87)

4. **Execution by Key-Holder Organizations** — Once funding amounts are determined, the Key-Holder Organizations SHALL jointly execute payments from the Coinholder-Controlled Fund using their k-of-n multisignature scheme (as defined in ZIP 1016). All transactions MUST be approved by the required threshold of signatures and published for community transparency.  
5. **Transparency Reporting** — The Administrator or its delegate MUST publish a summary report after each cycle detailing the funds available, amounts disbursed per proposal, any partial allocations or deferrals, and the hashes of executed transactions.  

This approach ensures that coinholder preferences are respected while maintaining a clear, deterministic process for allocating limited treasury resources and executing disbursements securely through the Key-Holder Organizations.  

---

### 6. Quarterly Timeline (Example)

| Phase | Approximate Duration | Description |
|:--|:--|:--|

> | **Public Call for Proposals** | ~ 1 month | The Administrator or its delegate announces an open call via community channels. Applicants prepare and submit completed-work proposals for consideration. |

Author: Addressed, please see [97efc59](https://github.com/zcash/zips/pull/1112/commits/97efc598c945c5a4a64bfbf30f3a7d36f0d90d87)

> | **Review and Discussion Period** | ≥ 30 days | Proposals are publicly discussed and evaluated. The Administrator or its delegate reviews submissions for eligibility and publishes a final list of qualified proposals. |

Reviewer: ```suggestion
| **Review and Discussion Period** | ≥ 30 days | Proposals are publicly discussed and evaluated. The Administrator or its delegate MUST review submissions for eligibility and publish a final list of qualified proposals. |
```

Author: Addressed, please see [97efc59](https://github.com/zcash/zips/pull/1112/commits/97efc598c945c5a4a64bfbf30f3a7d36f0d90d87)

The process described in this ZIP should describe how proposals are submitted, how voting is performed, how voting outcomes are determined, and how disbursements are made.

The roles of the Administrator, Key-Holder Organizations, Voting Authorities, and coinholders are expected to be clearly defined and distinguishable.

This ZIP should describe Key-Holder responsibilities for custody, security, and legal compliance in a clear and understandable manner.

The design is constrained to remain compatible with ZIP 1016 and with the NU 6.1 funding specification that governs key custody.

The process described in this ZIP is intended to operate on a quarterly cycle, with recurring rounds for proposal submission, review, voting, and disbursement.

This ZIP should not modify other Zcash funding programs or governance structures.

This ZIP must not require any changes to Zcash consensus.

# Specification

## Program Structure and Custody

The Coinholder Grants Program SHALL be administered by an organization to be designated prior to activation, referred to as the Administrator.

> Key custody and the n-of-m multisignature scheme controlling the Coinholder-Controlled Fund MUST follow the NU 6.1 funding specification referenced by ZIP 1016. Key-Holder Organizations MUST execute disbursements according to NU 6.1 [^zip-0271] and ZIP 1016’s veto provisions [^zip-1016-veto-process].

Reviewer 2: The first sentence is an overspecification. We can just rely on the fact that we are importing the definition of "Key-Holder Organizations" from ZIP 1016, and leave it to define how keys are managed.

#### Collecting Votes

The shielded voting protocol has a registration window corresponding to a defined block range. Eligibility is determined by Orchard notes created within this block range.

The Administrator MUST choose a registration window of at least 10 days.

The Administrator MUST publish voting guidance to coinholders. This guidance SHOULD include advising participants to move funds out of any wallet used for voting before entering a seed phrase into the Coin Voting 2.0 application, in order to reduce risk if the third-party application is compromised.

Votes MUST be submitted to a BFT voting chain operated by at least four independent Voting Authorities. The rationale for this threshold is that if at least two thirds of Voting Authorities are honest, vote suppression or manipulation becomes infeasible.

#### Counting Votes

The Administrator MUST collect and publish finalized tallies.

TBD: specify how shielded voting is counted.

#### Vote Auditing

Independent auditors MAY verify the results using the published election data.

> TBD: specify how shielded voting is audited.

Reviewer: Fill out this TBD

### Transparent Voting

#### Collecting Votes

Before accepting votes, the Administrator MUST publish an Orchard-only IVK designated for receiving vote messages and payments. The corresponding FVK MUST NOT be published.

To create a transparent vote, Voters:

- Create a signed message indicating their vote and transparent address using ECDSA over secp256k1 for the signature.
- Send the signed message along with a payment of at least 1 ZEC to the address associated with the published IVK.

TBD: finish the specification, including but not limited to:
- The signing protocol (just specifying a signature algorithm is insufficient).
- The address derivation to be used.
- The structure of transaction to be created (what kinds of outputs, how many, to what recipients, with what values).

#### Counting Votes

> TBD: specify how transparent votes are counted.

Reviewer: Shielded and transparent tallying should be specified in their respective sections, but that aside what you've written here is a high-level overview of the process. Some points that are missing (for the transparent side):
* What is the format of vote messages, and how are the signatures attached to the body of the vote message? 
* How are malformed vote messages to be interpreted?
* How is the coin weight for a transparent vote calculated? 
* How are signatures verified?

The Administrator MUST ignore submissions:

> - with payments below 1 ZEC.

Author: ```suggestion
- with a payment threshold chosen by the Administrator.
```

- where the Voter moves the submitted funds before voting concludes, as specified in the Quarterly Timeline below.

TBD: clarify specification of "voting concludes"; if you mean "at the end of the Coinholder Vote on Proposals phase" then say this instead of relying on the indirect definition.

#### Vote Auditing

The Administrator MUST publish IVK-based audit data and all received vote messages in order to allow verification.

TBD: specify how transparent votes are verified.

## Voting Process

The Administrator MUST publish relevant parameters for each voting method for each cycle, and provide sufficient information for independent auditing.

> TBD: specify the rest of the overall voting process.

Reviewer: Fill out this TBD. This should include:
- How to choose the parameters for each voting method 
- How the set of validators whose responses will be considered authoritative for determining the outcome of the vote is chosen
- Some of the timeline should probably be in this section - this should direct the administrator with respect to how to set the registration window, etc.
- Anything else that requires a choice by the administrator resulting from the completed specifications of the individual shielded and transparent voting mechanism sections.
- How the set of grants being voted on is translated into a set of individual elections.

The Administrator MUST publish a disbursement report detailing allocated amounts, partial disbursements, and executed transactions.

## Quarterly Timeline

| Phase                        | Approximate Duration  | Description |
|------------------------------|-----------------------|-------------|
| Public Call for Proposals    | ~ 1 month             | The Administrator or its delegate announces an open call via community channels. Applicants prepare and submit completed-work proposals for consideration. |
| Review and Discussion Period | ≥ 30 days             | Proposals are publicly discussed and evaluated. The Administrator or its delegate reviews submissions for eligibility and publishes a final list of qualified proposals. |
| Registration Period          | ≥ 10 days             | Coinholders register their balances for voting by creating eligible Orchard notes within a defined block range (as described in Section 4.1). |
| Coinholder Vote on Proposals | ~ 2 weeks             | Coinholders vote on eligible proposals using the process defined in Section 4. The Administrator or its delegate verifies and tallies the results. |

> | Vote Tallying                | After voting concludes | The Administrator or its delegate tallies all votes. For shielded voting, tallying is performed using the Coin Voting 2.0 Audit Tool [^zcash-vote-audit] to aggregate verified ballots submitted to the voting chain. For transparent voting, tallying is performed by verifying signed messages and corresponding transparent payments received at the published Orchard address using the IVK. Shielded and transparent results are combined into a unified tally prior to publication. |

Reviewer: Most of the content here should probably not be in this table, but should instead be in the relevant specification sections instead. 

The first sentence can stay here, the second sentence belongs in the shielded voting method section, the third in the transparent voting method section, and the final sentence belongs in Voting Process.

| Funding Disbursements        | —                     | After final tallies are published, the Administrator or its delegate coordinates with Key-Holder Organizations to execute approved payments from the multisig wallet. |

## Security Precautions

> Key-Holder Organizations MUST maintain custody according to the NU 6.1 funding specification and ZIP 1016.

Reviewer: Check that the NU6.1 reference is correct here.

Any theft or compromise of funds or key material MUST be reported promptly after mitigation.

# Rationale

TBD: add missing rationale for:
- Why the transparent voting FVK must not be published.
- The requirement for, and bound on, the payment amount for transparent voting.

> - The requirement that transparent voters not move their funds until voting concludes.

Reviewer: Fill in this TBD

# References

[^BCP14]: [Information on BCP 14 — "RFC 2119: Key words for use in RFCs to Indicate Requirement Levels" and "RFC 8174: Ambiguity of Uppercase vs Lowercase in RFC 2119 Key Words"](https://www.rfc-editor.org/info/bcp14)

[^protocol]: [Zcash Protocol Specification, Version 2025.6.1 [NU 6.1] or later](protocol/protocol.pdf)

[^protocol-networks]: [Zcash Protocol Specification, Version 2025.6.1 [NU 6.1]. Section 3.12: Mainnet and Testnet](protocol/protocol.pdf#networks)

[^zip-0271]: [ZIP 271: Deferred Dev Fund Lockbox Disbursement](zip-0271.md)

[^zip-1014]: [ZIP 1014: Establishing a Dev Fund for ECC, ZF, and Major Grants](zip-1014.rst)

[^zip-1015]: [ZIP 1015: Block Subsidy Allocation for Non-Direct Development Funding](zip-1015.rst)

[^zip-1016]: [ZIP 1016: Community and Coinholder Funding Model](zip-1016.md)

> [^zip-1016-veto-process]: [ZIP 1016: Community and Coinholder Funding Model — Veto Process](zip-1016.md#veto-process)

Reviewer: ```suggestion
[^zip-1016-disbursement-process]: [ZIP 1016: Community and Coinholder Funding Model — Disbursement Process](zip-1016.md#disbursementprocess)
```
